File: 2275-largest-combination-with-bitwise-and-greater-than-zero/2275-largest-combination-with-bitwise-and-greater-than-zero.java
class Solution {
    public int largestCombination(int[] arr) {
        
        int n=arr.length;int no=0;
        int max[]=new int[55];
        for(int i=0;i<n;i++)
        {
            no=arr[i];int f=1;int bit=1;
            while(no > 0)
            {
                if((no & f) > 0){
                    max[bit]++;
                }
                bit++;
                no=no >> 1;
            }
        }
        int ans=Integer.MIN_VALUE;
        for(int i=0;i<max.length;i++){
            ans=Math.max(ans,max[i]);
        }
        return ans;
    }
}

File: 2275-largest-combination-with-bitwise-and-greater-than-zero/2275-largest-combination-with-bitwise-and-greater-than-zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        vector<int> v(33);
        for(int i=0;i<candidates.size();i++)
        {
            int num=candidates[i];
            int j=0;
            while(num>0)
            {
                v[j]+=num%2;
                num/=2;
                j++;
            }
            
        }
        int maxi=INT_MIN;
        for(int i=0;i<32;i++)
        {
            
            maxi=max(maxi,v[i]);
        }
        return maxi;
        
        
    }
};